### PR TITLE
Move receiver and rssi configuration to receiver tab

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #81

### CODE REVIEW

**Comment by haslinghuis** [comment]
Not sure how the gulpfile.js file slipped in. This is a fresh clone in a new folder???

**Comment by haslinghuis** [comment]
Two problems remain after moving stuff.

1.  feature is filled with the RX stuff but the list does not show. 
2.  `EnabledDescription Feature` row is gone after fixing css (all of configuration css is in here until I find the solution)  - but no description after `RSSI_ADC`

Screenshot from 2021-01-14 02-02-37

Should display as:

Screenshot from 2021-01-14 02-26-38

There is no reboot in receiver tab. Changing receiver or enabling RSSI feature would require a reboot I guess?
We could make the Save button dynamic, changing to reboot depending on changing features added in this PR.

File: src/css/tabs/receiver.css [hunk 1]
```diff
@@ -140,6 +140,36 @@
     background-color: #14407a;
 }
 
+/* from configuration tab */
```
Comment (McGiverGim) [inline src/css/tabs/receiver.css:143]: Is better to remove the comment or change from something more "general" like "rx configuration". Any future user will not know nothing about what was in the configuration tab and what not and can confuse them.

File: src/css/tabs/receiver.css [hunk 2]
```diff
@@ -140,6 +140,36 @@
...
+    margin-top: 2px;
+}
+
+.tab-receiver dl.features dd {
+    margin: 0 0 0 20px;
+    height: 18px;
+    line-height: 18px;
+}
+
+.tab-receiver .serialRXBox, .spiRxBox {
```
Comment (McGiverGim) [inline src/css/tabs/receiver.css:168]: Maybe this must be:
```suggestion
.tab-receiver .serialRXBox, .tab-receiver .spiRxBox {
```

File: src/css/tabs/receiver.css [hunk 3]
```diff
@@ -559,6 +589,10 @@
         margin-bottom: 10px;
     }
 
+    select.features.rxMode, select.serialRX {
```
Comment (McGiverGim) [inline src/css/tabs/receiver.css:592]: This does not need the `.tab-receiver` to maintain consistence with the other styles? If it is a general `feature` style, maybe is better to go in a common css file. Is this if only for this tab, is better to add the `.tab-receiver` to go with the others styles.
```suggestion
    .tab-receiver select.features.rxMode, .tab-receiver select.serialRX {
```

File: src/css/tabs/receiver.css [hunk 4]
```diff
@@ -577,3 +611,284 @@
...
+    .tab-receiver .rssi td:nth-child(2) {
+        width: auto;
+    }
+    .tab-receiver .rxFailsafe td:nth-child(2) {
+        width: auto;
+    }
+    .tab-receiver .gps td:nth-child(2) {
+        width: auto;
+    }
+    select.features.rxMode, select.serialRX, .dshotbeacon > td > div.select > div, .dshotbeacon select.dshotBeeperBeaconTone {
```
Comment (McGiverGim) [inline src/css/tabs/receiver.css:888]: The same than before about adding the `.tab-receiver` to the style.

**Review by McGiverGim — COMMENTED** [review]
First of all thanks for all this PRs that make the Configurator easier and more intuitive.
Apart from the gulpfile that you have detected...
Only some comments about the css. 
What I see is that it has A LOT of new styles. Are all of them needed for that? Usually the major part of styles must be in the common css, only some differences if needed must be here, but I see a long list of them.
But the CSS is mesh of updates that maybe we can never clean so maybe this is the simple way to move this :)

**Comment by haslinghuis** [comment]
Fixed previous issues.
Only need to add 'Save and Reboot' (depending on changing rssi and serial configuration changes) and cleanup configuration.

**Comment by haslinghuis** [comment]
Okay. Because configuration does `Save and Reboot` and receiver tab only does `Save`, I've show the `Save and Reboot` instead of `Save` button only when changing receiver mode, serial receiver provider or RSSI_ADC.

**Comment by haslinghuis** [comment]
Rebased

**Comment by McGiverGim** [comment]
Is this out of draft? Sometimes we don't review the code because we are waiting until some kind of "final form" is ready...

**Comment by haslinghuis** [comment]
It's ready but it needs clean up , but I don't want conflict with #2356

**Comment by haslinghuis** [comment]
Fixed the dark side too :dark_sunglasses:

**Review by mikeller — APPROVED** [review]

**Comment by mikeller** [comment]
Looks good now.

Potentially one future improvement would be to move both ADC and RC channel based RSSI into the same panel, and only allow one to be selected at any time:

image

**Pull request merged**